http/README.md

```diff
@@ -0,0 +1,135 @@
+
+When this document says *HTTP* it is generally referring to *HTTP semantics*.
+
+## Interoperability with existing HTTP systems
+
+A goal of this spec is to allow libp2p to be able to interoperate with existing HTTP servers and clients. Care is taken in this document to not introduce anything that would break interoperability with existing systems.
+
+## HTTP Transport
+
+Nodes MUST use HTTPS (i.e., they MUST NOT use plaintext HTTP). It is RECOMMENDED to use HTTP/2 and HTTP/3.
+
+Nodes signal support for their HTTP transport using the `/http` component in
+their multiaddr. E.g., `/dns4/example.com/tls/http`. See the [HTTP multiaddr
+component spec](https://github.com/libp2p/specs/pull/550) for more details.
+
+## Namespace
+
+libp2p does not squat the global namespace. libp2p application protocols can be
+discovered by the [well-known resource](https://www.rfc-editor.org/rfc/rfc8615)
+`.well-known/libp2p/protocols`. This allows server operators to dynamically change the
+URLs of the application protocols offered, and not hard-code any assumptions how
+a certain resource is meant to be interpreted.
+
+```json
+
+{
+    "protocols": {
+        "/kad/1.0.0": {"path": "/kademlia/"},
+        "/ipfs/gateway": {"path": "/"},
+    }
+}
+```
+
+The resource contains a mapping of application protocols to a URL namespace. For
+example, this configuration file would tell a client
+
+1. That the Kademlia application protocol is available with prefix `/kademlia`
+and,
+2. The [IPFS Trustless Gateway API](https://specs.ipfs.tech/http-gateways/trustless-gateway/) is mounted at `/`.
+
+It is valid to expose a service at `/`. It is RECOMMENDED that implementations facilitate the coexistence of different service endpoints by ensuring that more specific URLs are resolved before less specific ones. For example, when registering handlers, more specific paths like `/kademlia/foo` should take precedence over less specific handler, such as `/`.
+
+## Peer ID Authentication
+
+When using the HTTP Transport, peer id authentication is optional. You only pay
+for it if you need it. This benefits use cases that don’t need peer
+authentication (e.g., fetching content addressed data) or authenticate some
+other way (not tied to libp2p peer ids).
+
+Specific authentications schemes for authenticating Peer IDs will be defined in
+a future spec.
+
+## Using HTTP semantics over stream transports
+
+Application protocols using HTTP semantics can run over any libp2p stream transport. Clients open a new stream using `/http/1.1` as the protocol identifer. Clients encode their HTTP request as an HTTP/1.1 message and send it over the stream. Clients parse the response as an HTTP/1.1 message and then close the stream. Clients SHOULD NOT pipeline requests over a single stream. Servers SHOULD set the [`Connection: close` header](https://datatracker.ietf.org/doc/html/rfc9112#section-9.6) to signal to clients that this is not a persistent connection.
```

Comment:
Is there a way for a multiaddr to signal support for this?
 
In most cases it doesn't make sense to speak http/1.1 over quic-v1 as specified in this section because you can simply do h3 over quic-v1 which is represented by the multiaddr:
`/ip4/1.2.3.4/udp/42/quic-v1/http`

Is there a way to specify that the node also speaks http/1.1(the variant specified in this section) over quic libp2p streams?